extension/jaegerremotesampling/internal/http_test.go

```diff
@@ -13,7 +13,7 @@ import (
 	"net/url"
 	"testing"
 
-	"github.com/jaegertracing/jaeger/thrift-gen/sampling"
+	"github.com/jaegertracing/jaeger/proto-gen/api_v2"
```

Comment:
I assume it does make sense to keep both implementations for now and add something like a feature flag to enable it? wdyt @codeboten @jpkrohling

Reply:
Makes sense to me 👍🏻

Reply:
Support enabling the new implementation via feature flag first, then make it the default on the following version, then remove the ability to use the old implementation in the next version before removing the old implementation altogether

Reply:
Any idea how to make the feature flag work without coping the entire protobuf definition from v1.41.0?

Reply:
Is it possible to use a replace statement to lock the version to the breaking version changes, and then within the next release(s) follow the feature gate rollout and then remove the replace?

Reply:
From my understanding no. Since the interface changes with the version bump. I think the easierst way would be to move the entire extension into a `deprecated` subfolder and put the feature flag into the factory. Its not nice, but since stuff gets removed soon..

Reply:
I don't see any other approach that would help here, just need to deprecated package doesn't down grade the version used in the main component.